addons/stock_dropshipping/models/sale.py

```diff
@@ -69,3 +69,14 @@ def _purchase_service_prepare_order_values(self, supplierinfo):
             res['dest_address_id'] = self.order_id.partner_shipping_id.id
             res['picking_type_id'] = dropship_operation.id
         return res
+
+    def _get_outgoing_incoming_moves(self, strict=True):
+        # moves for a return of an intercompany dropship are caught here
+        outgoing_moves, incoming_moves = super()._get_outgoing_incoming_moves(strict)
+        dropship_return_moves = outgoing_moves.filtered(
+            lambda m: m.origin_returned_move_id.picking_id.is_dropship
+        )
```

Comment:
I honestly think this could lead to unforeseen consequences in some cases :confused:

The main issue is that `move.location_dest_id._is_outgoing()` is `True` whether it's the original move or the return.
I wonder if it could be filtered out in original method near `(move.origin_returned_move_id and move.to_refund)`... But I didn't look around enough to know when that condition is useful (return of a return maybe?)